packages/router/src/grpc/view-protocol-server/note-by-commitment.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  NoteByCommitmentRequest,
+  NoteByCommitmentResponse,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { ViewReqMessage } from './router';
+import { ServicesInterface } from '@penumbra-zone/types';
+
+export const isNoteByCommitmentRequest = (req: ViewReqMessage): req is NoteByCommitmentRequest => {
+  return req.getType().typeName === NoteByCommitmentRequest.typeName;
+};
+
+export const handleNoteByCommitmentReq = async (
+  req: NoteByCommitmentRequest,
+  services: ServicesInterface,
+): Promise<NoteByCommitmentResponse> => {
+  const { indexedDb } = await services.getWalletServices();
+  if (!req.noteCommitment) return new NoteByCommitmentResponse();
+
+  // TODO  req.await_detection processing
```

Comment:
We have an issue for await detection somewhere. Can you add a todo in that issue to implement this?

Reply:
Yes, I was planning to add a TODO link to https://github.com/penumbra-zone/web/issues/233 after the merge